packages/nimble-components/src/table/specs/README.md

```diff
@@ -1,78 +1,276 @@
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type (i.e. do we need to create a `nimble-table-header`)
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+    -   Can clients customize header content?
+        -   How do we allow for icons to be used in the header (as shown in the visual design doc)
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS/design
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
 -   Grouping
--   Expand/collapse
--   Hierarchical data
+    -   Define interactive mechanism, if any, to provide grouping
+    -   Define table-level API for setting grouping
+    -   Define events raised when grouping changes
+    -   Describe how data hierarchy should work with grouping
+-   Sorting
+    -   Define the table-level API for setting sorting
+    -   Define events raised when sorting changes
+    -   Describe how sorting should with with hierarchical and/or grouped data
+-   Hierarchy
+    -   Define how hierarchy will be represented in the data
+        -   Attribute on table to represent parentId?
+    -   Define event/property APIs needed for dealing with lazily-loaded hierarchical data (possibly out of scope of initial release)
+    -   Describe the UI representation of hierarchical data (there should be a design doc to reference)
+-   Action Menu
+    -   Define how the action menu gets associated with a particular column
+    -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
+
+_Properties_
+
+Placeholder
+
+_Events_
+
+Placeholder
+
+### Anatomy
+
+_Slots_
+
+-   `slottedColumns` (Placeholder for column slots)
+-   `actionMenu` (Placeholder for action menu)
+
+### Security
+
+N/A
+
+### Angular integration
+
+Angular support should be accomplished through the typical directive patterns.
+
+There could be room for there to be Angular-centric implementations such as using Angular components for things like the Action Menu, or possibly a custom column. This section should be updated as needed.
+
+### Blazor integration
+
+Blazor support should be accomplished through the typical integration patterns.
+
+One aspect of note is that the data API for the `nimble-table` component will not be attribute-based, which means that the Blazor component will not be able to provide a binding-based association of its own data to that of the web component. Instead, we will need to leverage JSInterop to send the data to the web component, which is async by nature.
```

Comment:
mmm this might be too implementation detail-y or I might be misinterpreting. But I expect the user facing Blazor Component wrapper API is still going to be data binding based, any JS interop usage needed inside the blazor wrapper should be hidden from the user. Is that the expectation?

If anything we should make it our goal that JSInterop details are not exposed to users directly and it is an open question how that will be achieved until we investigate a bit more.

Reply:
So, my comment wasn't meant to imply that the JS Interop implementation would be exposed to the user in any way, but more of a signal to the developer of that, that that (sorry I had to grab hold of that opportunity to write 3 "that"s in a row) would be an implementation requirement (which isn't necessarily obvious).

My expectation is that the `Data` property is defined on the API surface like any other property would be (as a `Parameter`), but some implementation detail (be in the `setter` of that property, or some other override mechanism like `OnParameterSetAsync()` ) would handle the necessary JS Interop code. This means that there is no traditional binding of the `data` property on the web component in the Razor template (as we can't bind to properties...only attributes).

Reply:
Gotcha, makes sense 👍 Could phrase it in spec as "One aspect is that the Data property is not attribute based so the Binding from the blazor component wrapper Data property to the web component will require additional JSInterop considerations to preserve the expected API style" or something.

Reply:
Updated (to a variant of what you wrote).